content/docs/reference/state.md

```diff
@@ -1,140 +1,218 @@
+                    </div>
+                </li>
+                <li>
+                    <div class="mb-2">
+                        To use the <a href="#cloud-storage-or-remote">cloud storage or remote</a> option, run
+                        <code>pulumi login --cloud-url &lt;your-cloud-storage-url&gt;</code> so Pulumi stores your <span class="italic">checkpoint</span> files at your specified URL for your cloud storage object, also in a <code>.pulumi</code> directory. For example, <code>https://s3.console.aws.amazon.com/s3/buckets/my-pulumi-state-bucket>/.pulumi</code> where <span class="font-bold">my-pulumi-state-bucket</span> is the name of your S3 bucket. See the following links for details on cloud provider CLI and storage setup:
+                    </div>
+                        <div>
+                            <div class="mb-2">
+                                <span class="font-bold">AWS S3</span>:
+                                <span class="italic">See <a href="{{< relref "/docs/reference/clouds/aws/setup">}}">AWS Setup</a>
+                                and <a href="https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html">Working with Amazon S3 buckets</a>.</span>
+                            </div>
+                            <div class="mb-2">
+                                <span class="font-bold">Azure Blob</span>:
+                                <span class="italic">See <a href="{{< relref "/docs/reference/clouds/aws/setup">}}">Azure Setup</a> and Microsoft's <a href="https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-cli">Storage Blobs Quickstart</a>.</span>
+                        </div>
+                        <div class="mb-2">
+                            <span class="font-bold">Google Cloud Storage</span>:
+                            <span class="italic">See <a href="{{< relref "/docs/reference/clouds/aws/setup" >}}">GCP Setup</a>
+                            and Google's <a href="https://cloud.google.com/storage/docs/quickstarts">Cloud Storage Quickstarts</a>.</span>
+                        </div>
+                    </div>
+                </li>
+            </ul>
+        </div>
+    </div>
+</div>
+
+### Pulumi backend features
+
+The `pulumi.com` backend stores all checkpoint state securely, and the CLI always communicates directly with your cloud provider via client-side authentication. All state is encrypted in transit and at rest.
+
+Although backends are primarily responsible for storing state reliably, the `pulumi.com` backend also supports additional features that the filesystem or cloud backend does not. In particular, the `pulumi.com` backend supports keeping track of full
 deployment history (for auditing and rollback purposes), and supports concurrent locking so that you don't accidentally
```

Comment:
@CyrusNajmabadi: I see `backups`, `history`, and `stacks` as subdirectories on my S3's `.pulumi` directory. Does this mean that storing your checkpoint files remotely also gives you a full deployment history? What is the purpose of the `backups` directory?